File: TellDontAsk/AlarmsObjectOriented/CompositionRoot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using AlarmsObjectOriented.Adapters;
using AlarmsObjectOriented.Alarms;
using AlarmsObjectOriented.Criteria;
using AlarmsObjectOriented.Interfaces;
using NUnit.Framework;

namespace AlarmsObjectOriented
{
  public class CompositionRoot
  {
    private const string SecurityPhoneNumber = "11-222-1121";

    public static void Main()
    {
      var building = new Building(CreateAlarm());

      building.SomeoneEntered();
      building.AllClear();

      Console.WriteLine("========DUMP=========");
      building.Dump();

    }


    private static CompositeAlarm CreateAlarm()
    {
      return new CompositeAlarm
      (
        new TimedAlarm
        (
          new LoudAlarm(),
          AllOf
          (
            new NightCriterion(), 
            new WeekendCriterion()
          )
        ),
        new SilentAlarm
        (
          SecurityPhoneNumber
        )
      );
    }

    private static IEnumerable<TimeCriterion> AllOf(params TimeCriterion[] criterion)
    {
      return criterion;
    }
  }

  public class Spec
  {
    [Test]
    public void ShouldXxxxxxxxxxxx()
    {
      CompositionRoot.Main();
      Assert.Fail("Unfinished");
    }
  }
}

File: Java/AlarmsObjectOriented/src/main/java/CompositionRoot.java
import alarms.CompositeAlarm;
import alarms.LoudAlarm;
import alarms.SilentAlarm;
import alarms.TimedAlarm;
import criterias.NightCriteria;
import criterias.WeekendCriteria;
import interfaces.TimeCriteria;

import java.util.Arrays;
import java.util.Collection;

public class CompositionRoot {
  private final static String SecurityPhoneNumber = "11-222-1121";

  public static void main() {
    Building building = new Building(createAlarm());

    building.someoneEntered();
    building.allClear();

    System.out.println("========DUMP=========");
    building.dump();

  }


  private static CompositeAlarm createAlarm() {
    return new CompositeAlarm (
        new TimedAlarm (
            new LoudAlarm(),
            allOf (
                new NightCriteria(),
                new WeekendCriteria()
              )
          ),
        new SilentAlarm (
            SecurityPhoneNumber
          )
      );
  }

  private static Collection<TimeCriteria> allOf(TimeCriteria... criterias) {
    return Arrays.asList(criterias);
  }
}
